packages/atomic/src/components/atomic-folded-result-list/atomic-folded-result-list.tsx

```diff
@@ -0,0 +1,316 @@
+import {
+  Component,
+  h,
+  Element,
+  State,
+  Prop,
+  Listen,
+  Host,
+  Method,
+} from '@stencil/core';
+import {
+  ResultTemplatesManager,
+  buildResultTemplatesManager,
+  Result,
+  buildResultsPerPage,
+  ResultsPerPageState,
+  ResultsPerPage,
+  buildFoldedResultList,
+  FoldedResultList,
+  FoldedResult,
+  FoldedResultListState,
+  FoldedResultListProps,
+} from '@coveo/headless';
+import {
+  Bindings,
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../utils/initialization-utils';
+import {
+  ResultDisplayDensity,
+  ResultDisplayImageSize,
+  getResultDisplayClasses,
+} from '../atomic-result/atomic-result-display-options';
+import {TemplateContent} from '../atomic-result-template/atomic-result-template';
+import {once} from '../../utils/utils';
+import {updateBreakpoints} from '../../utils/replace-breakpoint';
+
+/**
+ * The `atomic-folded-result-list` component is responsible for displaying query results by applying one or more result templates.
+ *
+ * @internal
+ * @part result-list - The element containing every result of a result list
+ * @part result-list-grid-clickable - The clickable result when results are displayed as a grid
+ * @part result-table - The element of the result table containing a heading and a body
+ * @part result-table-heading - The element containing the row of cells in the result table's heading
+ * @part result-table-heading-row - The element containing cells of the result table's heading
+ * @part result-table-heading-cell - The element representing a cell of the result table's heading
+ * @part result-table-body - The element containing the rows of the result table's body
+ * @part result-table-row - The element containing the cells of a row in the result table's body
+ * @part result-table-cell - The element representing a cell of the result table's body
+ */
+@Component({
+  tag: 'atomic-folded-result-list',
+  styleUrl: 'atomic-folded-result-list.pcss',
+  shadow: true,
+})
+export class AtomicFoldedResultList implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  private resultList!: FoldedResultList;
+  public resultsPerPage!: ResultsPerPage;
+  private resultTemplatesManager!: ResultTemplatesManager<TemplateContent>;
+
+  @Element() private host!: HTMLDivElement;
+
+  @BindStateToController('resultList')
+  @State()
+  private resultListState!: FoldedResultListState;
+
+  @BindStateToController('resultsPerPage')
+  @State()
+  private resultsPerPageState!: ResultsPerPageState;
+
+  @State() public error!: Error;
+  @State() private templateHasError = false;
+
+  /**
+   * TODO:
+   */
+  @Prop({reflect: true}) public foldingCollection?: string;
```

Comment:
Since the wording in Headless is `collectionField`, `parentField`, `childField`, we should respect the same nomenclature in Atomic.

That way, if someone reads or search our documentation, they will always find the same name for everything, versus subtle variations between each projects.